spec/support/contexts/with_active_record.rb

```diff
@@ -0,0 +1,115 @@
+require 'active_record'
+
+configurations = {
+  'default' => {
+    'adapter'  => 'sqlite3',
+    'database' => ':memory:'
+  }
+}
+ActiveRecord::Base.configurations = configurations
+ActiveRecord::Base.establish_connection :default
+
+##
+# Creates a database schema for the test, in memory, builds up models, and
+# removes them once finished.
+#
+# Example:
+#
+#  ```
+#  describe "Book#where", :with_active_record_schema
+#  end
+#  ```
+#
+shared_context 'with active_record', :with_active_record do
```

Comment:
Is it possible to just load AR/models in `spec_helper` like most other AR plugins?

Reply:
Have an example of this?  Not fully sure what you are suggesting here.

I admit that some of what I am doing here is probably overly complicated (because the concept for this was pulled from a different project where I was trying to have variable relations with the models (for example:  if there was a `comments` table requested for this this context, include the `has_many :comments` in the `Book` model definition).

Not really needed for what I am doing here though, but some of the weirdness from that remained (defining classes inline for example).  Could live without that here, probably.

Reply:
I guess I meant that in spec_helper, it would always initialize active record with a sqlite db (ala `:memory:`) and "assume" that we need a database.

Reply:
Right, this is the only file where testing against `active_record` makes sense.